components/TwitterTestimonials.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <v-card
+    class="mx-auto tweet-widget"
+    :class="$vuetify.breakpoint.mobile ? 'mobile' : ''"
+    color="white"
+  >
+    <v-carousel
+      v-model="activeTweetIndex"
+      hide-delimiters
+      :show-arrows="!$vuetify.breakpoint.mobile"
+      light
+    >
+      <v-carousel-item v-for="(tweet, i) in tweets" :key="tweet.id">
+        <tweet :tweet="tweets[i]" />
+      </v-carousel-item>
+    </v-carousel>
+    <tweet-carousel-delimiter
+      v-show="$vuetify.breakpoint.mobile"
+      :steps="3"
+      :index="activeTweetIndex"
+    />
+  </v-card>
+</template>
+<script>
+import testimonialsData from "../data/testimonials.json";
+import tweetFixtures from "../fixtures/twitterTweetsApiFixture.json";
+
+const API_URL =
+  "https://api.twitter.com/2/tweets?user.fields=profile_image_url,username&expansions=author_id&ids=";
+const DEV_MODE = "DEV_MODE";
+
+const token = process.env.TWITTER_API_BEARER_TOKEN || DEV_MODE;
+const options = {
+  headers: { Authorization: `Bearer ${token}` },
+  mode: "no-cors",
+};
+
+export default {
+  data() {
+    return {
+      tweets: [],
+      activeTweetIndex: 0,
+      isLoaded: false,
+    };
+  },
+  methods: {
+    createUserMap: function (apiUserData) {
+      /* Maps users by id */
+      const userMap = {};
+      for (const user of apiUserData) {
+        userMap[user.id] = user;
+      }
+      return userMap;
+    },
+    annotateTweetsWithUsers: function (apiTweetData, userMap) {
+      /* The twitter API separates user information and tweet content. This method recombines, overwriting fields as necessary */
+      let annotated = [];
+      for (const tweet of apiTweetData) {
+        const annotatedTweet = Object.assign(tweet, {
+          name: userMap[tweet.author_id].name,
+          username: userMap[tweet.author_id].username,
+          profile_image_url: userMap[tweet.author_id].profile_image_url,
+          profile_image_url_original: userMap[
+            tweet.author_id
+          ].profile_image_url.replace(/_normal\./, "."),
+        });
+        annotated.push(annotatedTweet);
+      }
+      return annotated;
+    },
+  },
+  async fetch() {
+    /* Automatically called on component instantiation */
+
+    let json = "";
+    if (token === DEV_MODE) {
+      // eslint-disable-next-line
+      console.log(
+        "Twitter instantiated using dev mode; sample tweets will be loaded from fixtures"
+      );
+      json = tweetFixtures;
+    } else {
+      json = await fetch(`${API_URL}${testimonialsData.tweets}`, options).then(
```

Comment:
you shouldn't need `.then()` since you're using `await`

Reply:
right, was trying not to make additional throw away vars, as this would otherwise need something like

```
const res = await fetch(`${API_URL}${testimonialsData.tweets}`, options);
json = res.json();
```

Not attached though.

Reply:
would definitely change since using `await` with `.then` are two conflicting patterns. 
Personally, I prefer the extra var since variable names can help explain what's happening. (you and I might be able to understand, but not everyone in the codebase has as much exp 🏅)

Reply:
roger roger